app/celer-sim/Runner.cc

```diff
@@ -361,23 +366,17 @@ void Runner::build_events(RunnerInput const& inp)
         std::mt19937 rng;
         auto generate_event = PrimaryGenerator<std::mt19937>::from_options(
             core_params_->particle(), inp.primary_gen_options);
-        auto event = generate_event(rng);
-        while (!event.empty())
-        {
-            append(event);
-            event = generate_event(rng);
-        }
+        read_events([&generate_event, &rng] { return generate_event(rng); });
+    }
+    else if (ends_with(inp.event_filename, ".root"))
+    {
+        read_events(
+            RootEventReader(inp.event_filename, core_params_->particle()));
```

Comment:
Since now we read from both HepMC3 and ROOT, we should update the doxygen comment of this function to something like
```
Read events from a HepMC3/ROOT input file or build using a primary generator.
```